src/libxtp/molden.cc

```diff
@@ -230,7 +231,12 @@ void Molden::parseMoldenFile(const std::string& filename,
       boost::trim(units);
       XTP_LOG(Log::error, _log)
           << "Reading atoms using " << units << " units." << std::flush;
-      line = readAtoms(orbitals.QMAtoms(), units, input_file);
+      // only update atoms if they are not yet loaded
+      if (orbitals.QMAtoms().size() == 0) {
+        line = readAtoms(orbitals.QMAtoms(), units, input_file);
+      } else {  // we need get the next line to continue
+        std::getline(input_file, line);
+      }
```

Comment:
what happens if orca updated the coordinates? I do not fully understand this check

Reply:
Good question, what happens is that we still use the "xtp" coordinates. The reason for this if statement originally was that without it, it would just append the molden atoms to the orb object, which would mean that you get the same atoms twice in the orbitals. To prevent this I added the if statement. In principle orca shouldn't change the coordinates because that would make it different from our xtp dft implementation which seems undesirable. As an alternative I could delete the existing atoms and reread them, but I didn't see the benefit, moreover after the 8th decimal place or so we might lose a bit of accuracy by writing them to and reading them from text.

Reply:
I do not know, before we read in the coordinates if orca uses the `optimize` keyword. I think it is better to read them in and remove the old atoms. Especially because the xyz files also do not really have 7 digits.

Reply:
Okay